server/service/logging_campaigns.go

```diff
@@ -0,0 +1,75 @@
+package service
+
+import (
+	"context"
+	"time"
+
+	"github.com/kolide/fleet/server/contexts/viewer"
+	"github.com/kolide/fleet/server/kolide"
+	"github.com/kolide/fleet/server/websocket"
+)
+
+func (mw loggingMiddleware) NewDistributedQueryCampaign(ctx context.Context, queryString string, hosts []uint, labels []uint) (*kolide.DistributedQueryCampaign, error) {
+	var (
+		loggedInUser = "unauthenticated"
+		err          error
+	)
+	vc, ok := viewer.FromContext(ctx)
+	if ok {
+		loggedInUser = vc.Username()
+	}
+
+	defer func(begin time.Time) {
+		_ = mw.logger.Log(
+			"method", "NewDistributedQueryCampaign",
+			"err", err,
+			"user", loggedInUser,
+			"numHosts", len(hosts),
```

Comment:
This would show how many hosts are targeted by their ID, but not how many unique hosts are targeted through labels and IDs. You could make use of https://github.com/kolide/fleet/blob/a856fd5c0be21044dffd743e6cfbe2764424ad5a/server/service/service_targets.go#L30 to get the actual count.